lib/widgets/recent_dm_conversations.dart

```diff
@@ -138,3 +149,61 @@ class RecentDmConversationsItem extends StatelessWidget {
           ]))));
   }
 }
+
+class _NewDmButton extends StatefulWidget {
+  const _NewDmButton();
+
+  @override
+  State<_NewDmButton> createState() => _NewDmButtonState();
+}
+
+class _NewDmButtonState extends State<_NewDmButton> {
+  bool _pressed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final fabBgColor = _pressed
+      ? designVariables.fabBgPressed
+      : designVariables.fabBg;
+    final fabLabelColor = _pressed
+      ? designVariables.fabLabelPressed
+      : designVariables.fabLabel;
+
+    return GestureDetector(
+      onTap: () => showNewDmSheet(context),
+      onTapDown: (_) => setState(() => _pressed = true),
+      onTapUp: (_) => setState(() => _pressed = false),
+      onTapCancel: () => setState(() => _pressed = false),
+      child: AnimatedContainer(
+        duration: const Duration(milliseconds: 100),
+        curve: Curves.easeOut,
+        padding: const EdgeInsetsDirectional.fromSTEB(16, 12, 20, 12),
+        decoration: BoxDecoration(
+          color: fabBgColor,
+          borderRadius: BorderRadius.circular(28),
+          boxShadow: [BoxShadow(
+            color: designVariables.fabShadow,
+            blurRadius: _pressed ? 12 : 16,
+            spreadRadius: 0,
```

Comment:
nit: leave out argument that equals the boring default
```suggestion
          boxShadow: [BoxShadow(
            color: designVariables.fabShadow,
            blurRadius: _pressed ? 12 : 16,
```